doc/book/examples/sources/basics/custom-transfer.move

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module 0x0::restricted_transfer {
+    use sui::tx_context::{Self, TxContext};
+    use sui::balance::{Self, Balance};
+    use sui::coin::{Self, Coin};
+    use sui::id::VersionedID;
+    use sui::transfer;
+    use sui::sui::SUI;
+
+    /// For when paid amount is not equal to the transfer price.
+    const EWrongAmount: u64 = 0;
+
+    /// An object that marks a property ownership
```

Comment:
Might be worthwhile to say something about who would have the permission to mint these and how dedup might be handled.